flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/logical/operators.scala

```diff
@@ -236,6 +236,32 @@ case class Aggregate(
   }
 }
 
+case class SetMinus(left: LogicalNode, right: LogicalNode, all: Boolean) extends BinaryNode {
+  override def output: Seq[Attribute] = left.output
+
+  override protected[logical] def construct(relBuilder: RelBuilder): RelBuilder = {
+    left.construct(relBuilder)
+    right.construct(relBuilder)
+    relBuilder.minus(all)
+  }
+
+  override def validate(tableEnv: TableEnvironment): LogicalNode = {
+    val resolvedMinus = super.validate(tableEnv).asInstanceOf[SetMinus]
+    if (left.output.length != right.output.length) {
+      failValidation(s"Set minus two table of different column sizes:" +
+        s" ${left.output.size} and ${right.output.size}")
+    }
+    val sameSchema = left.output.zip(right.output).forall { case (l, r) =>
+      l.resultType == r.resultType && l.name == r.name }
```

Comment:
I don't think it makes much sense to perform minus between sets {(1, 2, 3), (4, 5, 6)} and {(1, 2, 3, 10, 20), (4, 5, 6, 50, 60)}. Such operation would likely be a sign of a user error and it seems to me that it's better to throw an exception than silently do something surprising. Therefore I don't think that testMinusDifferentFieldsNames should pass.

As for a case when we have different names but identical field count and type I don't see any difference between a union and minus. It seems to me that both union and minus are just set operations on tuples and technically both can work in case when we have tables with different field names.

@fhueske What do you think about it?